index.js

```diff
@@ -287,6 +287,7 @@ NYC.prototype._handleSourceMap = function (cacheDir, code, hash, filename) {
 
 NYC.prototype._handleJs = function (code, filename) {
   var relFile = path.relative(this.cwd, filename)
+  filename = path.resolve(this.cwd, relFile) // ensure the path has correct casing (see https://github.com/istanbuljs/nyc/issues/269)
```

Comment:
@JimmyBoh I'd also be tempted to add a reference to the underlying issue in node.js, so that we can retire the filename logic in the future.